### PR TITLE
added support for absolute filenames

### PR DESCRIPTION
**Pull request by inmarelibero**
I saw the other PR stuck for 4 months. personally I don't need that feature now, but it could be useful maybe.

### CODE REVIEW

**Comment by inmarelibero** [comment]
maybe

```
// calculate absolute or relativa path
if (array_key_exists('is_absolute_path', $options) && true === $options['is_absolute_path']) {
    $this->fullFilename = $filename;
} else {
    $this->fullFilename = $this->baseDir.'/'.$filename;
}
```

should go into `parseOptions()`

**Comment by igorw** [comment]
Will not add this unless there is a real use case for it.

**Comment by inmarelibero** [comment]
ok, good for me

**Comment by kbond** [comment]
I have need of this feature

File: DependencyInjection/IgorwFileServeExtension.php [hunk 1]
```diff
@@ -18,5 +18,6 @@ public function load(array $configs, ContainerBuilder $container)
         $config = $this->processConfiguration($configuration, $configs);
         $container->setAlias('igorw_file_serve.response_factory', sprintf('igorw_file_serve.response_factory.%s', $config['factory']));
         $container->setParameter('igorw_file_serve.base_dir', $config['base_dir']);
+        $container->setParameter('igorw_file_serve.is_absolute_path', $config['is_absolute_path']);
```
Comment (kbond) [inline DependencyInjection/IgorwFileServeExtension.php]: why is this here?

**Comment by kbond** [comment]
Can we not just check if the filename given to `create` is readable to see if it is absolute (instead of setting an option)?

File: DependencyInjection/IgorwFileServeExtension.php (hunk 1 above)
Comment (inmarelibero) [inline DependencyInjection/IgorwFileServeExtension.php]: because it needs to bind the the parameters in config.yml, no?

Comment (kbond) [inline DependencyInjection/IgorwFileServeExtension.php]: There is no `is_absolute_path` bundle config

Comment (inmarelibero) [inline DependencyInjection/IgorwFileServeExtension.php]: well I need to retake this PR before going further. you're right anyway

**Comment by inmarelibero** [comment]
@kbond  imo I don't think it's a good pattern, it seems a not 100% clear behavior for me

**Comment by kbond** [comment]
ok, fair enough

**Comment by igorw** [comment]
Yeah, implicitly allowing absolute paths might potentially be dangerous, having an option is a good thing imo. What's the status of this PR? Is it mergeable?

**Comment by inmarelibero** [comment]
no it's not mergeable right now. I'll retake the PR and prepare it as soon as possible. @kbond spotted an error and I'm going to fix it. @kbond you are welcome to review the state of the PR

**Comment by inmarelibero** [comment]
@igorw @kbond ok it can be merged for me.
maybe it should be tested better, but the modifications at the code are pretty straightforward

**Comment by kbond** [comment]
I have tested your fork with my project and it works perfectly.  I would say it's mergeable.

**Comment by igorw** [comment]
Merged with amendments and tagged v1.0.1. I renamed the option to `absolute_path`.

**Comment by inmarelibero** [comment]
fair enough, thanks!

**Comment by igorw** [comment]
Thanks for the patch and testing btw!

**Comment by kbond** [comment]
@igorw the refactor broke it, see #10

heh, this was a perfect example of why tdd is a good thing

**Comment by igorw** [comment]
Damn, I failed. Thanks for the fixup.
